### PR TITLE
Fix/#118

### PR DESCRIPTION
**Pull request by mlindauer**
@mfeurer please check this PR. I would like to remove the aggregate function in another PR since it will be otherwise mess up this PR.

* ADD two new runhistory functions: compute_all_costs() and incremental_update_cost
* FIX merge_foreign_data updated accordingly
* add unit tests

### CODE REVIEW

File: smac/runhistory/runhistory.py [hunk 1]
```diff
@@ -45,6 +46,7 @@ def __init__(self, aggregate_func):
         self._n_id = 0
 
         self.cost_per_config = {} # config_id -> cost
+        self.runs_per_config = {} # config_id -> number of runs
```
Comment (mfeurer) [inline smac/runhistory/runhistory.py]: Could you please add a comment that this is necessary to calculate the moving average?

File: smac/runhistory/runhistory.py [hunk 2]
```diff
@@ -103,6 +107,50 @@ def update_cost(self, config):
...
+            Arguments
+            --------
+            config: Configuration
+                configuration to update cost based on all runs in runhistory
+            cost: float
+                cost of new run of config
+        '''
+        
+        config_id = self.config_ids[config]
+        n_runs = self.runs_per_config.get(config_id,0)
```
Comment (mfeurer) [inline smac/runhistory/runhistory.py]: Please add whitespaces after commas.

File: smac/runhistory/runhistory.py [hunk 3]
```diff
@@ -103,6 +107,50 @@ def update_cost(self, config):
...
+        '''
+            computes the cost of all configurations from scratch
+            and overwrites self.cost_perf_config and self.runs_per_config accordingly;
+            Since we iterate over the runhistory for every configuration,
+            this function can be quite expensive.
+            
+            Arguments
+            ---------
+            instances: typing.List[str]
+                list of instances; if given, cost is only computed with wrt to this instance set
```
Comment (mfeurer) [inline smac/runhistory/runhistory.py]: with and wrt are redundant

File: smac/runhistory/runhistory.py (hunk 1 above)
Comment (mlindauer) [inline smac/runhistory/runhistory.py]: Thanks. Fixed

File: smac/runhistory/runhistory.py (hunk 3 above)
Comment (mlindauer) [inline smac/runhistory/runhistory.py]: Thanks. Fixed

File: smac/runhistory/runhistory.py (hunk 2 above)
Comment (mlindauer) [inline smac/runhistory/runhistory.py]: Thanks. I applied an automatic pep8 conversion now :-)

**Pull request merged**